src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,66 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final String DEFAULT_COLOR = "White";
+    public static final int DEFAULT_RADIUS = 10;
+    private static final int FIGURE_TYPES = 5;
+    private static final double DIMENSION_RANGE = 10.0;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(FIGURE_TYPES);
+        String color = colorSupplier.getRandomColor();
+        switch (figureType) {
+            case 0:
+                return createSquare(color);
+            case 1:
+                return createRectangle(color);
+            case 2:
+                return createRightTriangle(color);
+            case 3:
+                return createCircle(color);
+            default:
+                return createIsoscelesTrapezoid(color);
+        }
+    }
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. For instance, in the 'createSquare' method, you should create a variable for the side length before passing it to the constructor.